app/models/protocol.rb

```diff
@@ -118,21 +123,25 @@ def has_human_subject_info?
     self.research_types_info.try(:human_subjects) || false
   end
 
-  validate :existing_rm_id,
-    if: -> record { Setting.get_value("research_master_enabled") && !record.research_master_id.nil? }
-
-  validate :unique_rm_id_to_protocol,
-    if: -> record { Setting.get_value("research_master_enabled") && !record.research_master_id.nil? }
-
   def self.rmid_status
     begin
       HTTParty.get(Setting.get_value("research_master_api") + 'research_masters.json', headers: {'Content-Type' => 'application/json', 'Authorization' => "Token token=\"#{Setting.get_value("rmid_api_token")}\""})
       return true
     rescue
+      @@rmid_server_down = true
       return false
     end
   end
 
+  def self.get_rmid(rmid)
+    begin
+      HTTParty.get("#{Setting.get_value('research_master_api')}research_masters/#{rmid}.json", headers: { "Content-Type" => "application/json", "Authorization" => "Token token=\"#{Setting.get_value('rmid_api_token')}\"" })
+    rescue
+      @@rmid_server_down = true
+      nil
+    end
+  end
```

Comment:
Returns the RMID as a JSON hash. I pulled this code from SPARCBiobank where we also validate RMIDs